### DISCUSSION TITLE
Read exactly X bytes from stream

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/runtime#67975] [Q&A] [answered]
I need to read blocks of data from a stream.  If I ask for X bytes, I should get exactly X bytes from the stream unless EOF is reached and then I should get whatever is left.

StreamReader has a ```ReadBlock``` method but it reads chars.  I need a method just like this but for bytes.

Is there a built in way to do that or do I have to manually call read over and over and advance my position in my buffer?

**Comment by gfoidl** [comment, marked as answer]
> Is there a built in way to do that or do I have to manually call read over and over and advance my position in my buffer?

Loop manually (as documented) or wait for https://github.com/dotnet/runtime/issues/16598.

**Reply by TonyValenti** [reply]
That is exactly what I need!
